skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -0,0 +1,596 @@
+import numpy as np
+from skimage import util
+cimport numpy as np
+cimport cython
+from libc.math cimport exp
+
+ctypedef np.float32_t DTYPE_t
```

Comment:
I haven't seen this pattern before... Is it here in case we later want to change the dtype to float64? If so, could we use a different alias from `DTYPE_t`? It's not very descriptive. I'm thinking something like `IMGDTYPE`.